include/k4arecord/record.hpp

```
@@ -0,0 +1,280 @@
+/** \file record.hpp
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ * Kinect For Azure SDK - C++ wrapper.
+ */
+#ifndef K4A_RECORD_HPP
+#define K4A_RECORD_HPP
+
+#include <k4a/k4a.hpp>
+#include <k4arecord/record.h>
+
+namespace k4a
+{
+
+/** \class record record.hpp
+ * Wrapper for \ref k4a_record_t
+ *
+ * Wraps a handle for a record object
+ *
+ * \sa k4a_record_t
+ */
+class record
+{
+public:
+    /** Creates a k4a::record from a k4a_record_t
+     * Takes ownership of the handle, i.e. you should not call
+     * k4a_record_close on the handle after giving it to the
+     * k4a::record; the k4a::record will take care of that.
+     */
+    record(k4a_record_t handle = nullptr) noexcept : m_handle(handle) {}
+
+    /** Moves another k4a::record into a new k4a::record
+     */
+    record(record &&other) noexcept : m_handle(other.m_handle)
+    {
+        other.m_handle = nullptr;
+    }
+
+    record(const record &) = delete;
+
+    ~record()
+    {
+        flush();
```

Think we should just move flush into the close function. I don't see a scenario where you don't want that called.

If we don't, then call to flush should be wrapped in a `if (m_handle){}` check

@xthexder, what do you think?

---

k4a_record_close() will already call flush internally, but since it's a void function, errors are ignored.
Flush was made accessible as a separate function so the user can flush periodically and verify data was written to disk.
If we're not handling flush errors, then there is no point to calling flush separately.

---

I see. I remove flush from this destructor.